### PR TITLE
[storage] fix the version update

### PR DESCRIPTION
**Pull request by areshand**
### Description
update the version while saving txn ws
### Test Plan
updated unit test and manual test with more iteration

### CODE REVIEW

File: storage/backup/backup-cli/src/backup_types/transaction/tests.rs [hunk 1]
```diff
@@ -120,19 +120,22 @@ fn end_to_end() {
...
-            .take(target_version as usize)
-            .map(|txn_to_commit| txn_to_commit.write_set().clone())
-            .collect::<Vec<_>>()
-    );
+
+    for (restore_ws, org_ws) in ouptputlist
+        .transactions_and_outputs
+        .iter()
+        .map(|(_, output)| output.write_set().clone())
+        .zip(
```
Comment (msmouse) [inline storage/backup/backup-cli/src/backup_types/transaction/tests.rs:128]: why the change?
This is worse because it doesn't assert both sequences have the same length (for example, if either of them is empty, this will pass).

**Review by msmouse — COMMENTED** [review]
how was it flaky.. not always failing ❓ 😂

**Comment by areshand** [comment]
> how was it flaky.. not always failing ❓ 😂

This is because the save_transaction is done per chunk. If the chunk only has 1 transaction, the test won't catch it. The chunk logic is also based on the record byte size. The test record can have very large size so the chunk can frequent just has 1 txn.

File: storage/backup/backup-cli/src/backup_types/transaction/tests.rs [hunk 2]
```diff
@@ -120,19 +120,21 @@ fn end_to_end() {
...
+    let iter_before = blocks
+        .iter()
+        .flat_map(|(txns, _li)| txns)
+        .take(target_version as usize)
+        .map(|txn_to_commit| txn_to_commit.write_set().clone());
+
+    for (restore_ws, org_ws) in iter_after.clone().zip(iter_before.clone()) {
+        assert_eq!(restore_ws, org_ws);
+    }
+    assert_eq!(iter_before.count(), iter_after.count());
```
[the remark this answers is not in the document]
Comment (areshand) [reply storage/backup/backup-cli/src/backup_types/transaction/tests.rs]: cool, updated
